### PR TITLE
Push the ActiveFedora dependency out of the interface for Preservatio…

### LINKED ISSUE #3178: Rewrite PreserveJob / SdrIngestService to use Cocina object

### PR DESCRIPTION
**Pull request by jcoyne**
…nMetadataExtractor

## Why was this change made? 🤔
Fixes #3178
Decoupling from ActiveFedora.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: spec/services/sdr_ingest_service_spec.rb [hunk 1]
```diff
@@ -113,17 +112,17 @@
 
   describe '.signature_catalog_from_preservation' do
     let(:druid) { 'druid:dd116zh0343' }
-    let(:dor_item) { instance_double(Dor::Item, pid: druid) }
+    let(:cocina) { instance_double(Cocina::Models::DRO, externalIdentifier: druid) }
```
Comment (jmartin-sul) [inline spec/services/sdr_ingest_service_spec.rb:115]: does this let var actually get referenced?  cruft, or me just not seeing where it's used?

**Review by jmartin-sul — APPROVED** [review]
one question about a possibly extraneous let var in a test, but otherwise LGTM.

File: spec/services/sdr_ingest_service_spec.rb (hunk 1 above)
Comment (jmartin-sul) [reply spec/services/sdr_ingest_service_spec.rb:115]: (feel free to merge if i'm off-base, or after removing the var if it actually is superfluous)

**Review by justinlittman — APPROVED** [review]

File: spec/services/sdr_ingest_service_spec.rb (hunk 1 above)
Comment (jcoyne) [reply spec/services/sdr_ingest_service_spec.rb:115]: It doesn't seem to be used. I'll remove it.

**Pull request merged**